### ISSUE TITLE
[bitnami/external-dns] Cannot use both `txtEncrypt.aesKey` and `cloudflare.secretName`

### ISSUE THREAD

**Issue by Felerius** [bitnami/charts#27828]
### Name and Version

bitnami/external-dns 8.0.2

### What architecture are you using?

None

### What steps will reproduce the bug?

Try to install the helm chart with the values I provided.

### Are you using any custom parameters or values?

```yaml
provider: cloudflare
cloudflare:
  secretName: cloudflare-token
txtEncrypt:
  enabled: true
  aesKey: some-random-key
```

### What do you see instead?

Installation fails with the following error:

```
cloudflare.apiKey is required if cloudflare.apiToken is not provided
```

### Additional information

The `external-dns.createSecret` template in [`_helpers.tpl`](https://github.com/bitnami/charts/blob/main/bitnami/external-dns/templates/_helpers.tpl) forces the secret defined in [`secret.yaml`](https://github.com/bitnami/charts/blob/main/bitnami/external-dns/templates/secret.yaml) to be created if the `txtEncrypt.aesKey` is provided. The cloudflare block in the latter seems to assume that it will only be reached if `cloudflare.secretName` isn't set and thus either `cloudflare.apiKey` or `cloudflare.apiToken` need to be set.

**Comment by carrodher** [comment]
Thank you for bringing this issue to our attention. We appreciate your involvement! If you're interested in contributing a solution, we welcome you to create a pull request. The Bitnami team is excited to review your submission and offer feedback. You can find the contributing guidelines [here](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md).

Your contribution will greatly benefit the community. Feel free to reach out if you have any questions or need assistance.

**Comment by gabricar-andela** [comment]
hi, can we reopen this? It's still an issue.

**Comment by luryus** [comment]
Please reopen, this is still valid. PR for a fix here: #32360

**Comment by carrodher** [comment]
Thank you for submitting the associated Pull Request. Your contribution is greatly appreciated!
